AlphaWallet/Core/BuyToken/CoinBase/CoinBase.swift

```diff
@@ -0,0 +1,38 @@
+//
+//  CoinBase.swift
+//  AlphaWallet
+//
+//  Created by Vladyslav Shepitko on 23.08.2022.
+//
+
+import Foundation
+import Combine
+
+final class CoinBase: SupportedTokenActionsProvider, BuyTokenURLProviderType {
+    var objectWillChange: AnyPublisher<Void, Never> {
+        return .empty()
+    }
+
+    var analyticsName: String { "CoinBase" }
+    let analyticsNavigation: Analytics.Navigation = .onCoinBase
+    var action: String {
+        return R.string.localizable.aWalletTokenBuyOnCoinBaseTitle()
+    } 
+
+    func url(token: TokenActionsIdentifiable, wallet: Wallet) -> URL? {
+        guard let platform = token.server.coinBasePlatform else { return nil }
+        return Constants.buyWitCoinBaseUrl(blockchain: platform, wallet: wallet).flatMap { URL(string: $0) }
+    }
+
+    func actions(token: TokenActionsIdentifiable) -> [TokenInstanceAction] {
+        return [.init(type: .buy(service: self))]
+    }
+
+    func isSupport(token: TokenActionsIdentifiable) -> Bool {
+        return token.server.coinBasePlatform != nil
```

Comment:
Maybe should just check server == .main instead?

Reply:
documentation says Avalanche is supported

Reply:
lets make it only for main for now